src/test/java/com/es/phoneshop/model/product/ArrayListProductDaoTest.java

```diff
@@ -1,57 +1,118 @@
 package com.es.phoneshop.model.product;
 
-import com.es.phoneshop.model.product.DAO.ArrayListProductDao;
-import com.es.phoneshop.model.product.DAO.ProductDao;
-import com.es.phoneshop.model.product.entity.Product;
+import com.es.phoneshop.DAO.CustomProductDao;
+import com.es.phoneshop.DAO.ProductDao;
+import com.es.phoneshop.model.entity.Product;
+import org.junit.After;
 import org.junit.Before;
 import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
 
 import java.math.BigDecimal;
 import java.util.Currency;
 import java.util.List;
+import java.util.NoSuchElementException;
 
 import static org.junit.Assert.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doAnswer;
 
-public class ArrayListProductDaoTest {
+@RunWith(MockitoJUnitRunner.class)
+public class ArrayListProductDaoTest {// i suppose service tests must be also written
     Currency usd = Currency.getInstance("USD");
-    private ProductDao productDao;
+    ProductDao productDao;
+    @Mock
+    ProductDao productDaoMock;
+    Product product;
+    List<Product> productList;
 
     @Before
     public void setup() {
-        productDao = new ArrayListProductDao();
+        productDao = CustomProductDao.getCustomProductDao();
+
+        product = new Product("test", "test", new BigDecimal(100), usd, 10, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Samsung/Samsung%20Galaxy%20S.jpg");
+
+        productList = CustomProductDao.getCustomProductDao().getProducts();
+        doAnswer(invocation -> {//mocking save
+            Product productToSave = invocation.getArgument(0);
+            productList.add(productToSave);
+            return null;
+        }).when(productDaoMock).save(any(Product.class));
+        productDaoMock.save(product);
+    }
+
+    @After
+    public void rearrange() {
+        productList.remove(product);
     }
 
     @Test
-    public void testFindProductsNoResults() {
-        Product product = productDao.save(new Product("test", "test", new BigDecimal(100), usd, 0, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Samsung/Samsung%20Galaxy%20S.jpg"));
-        assertTrue(product.getId() > 0);
+    public void givenAProductWithZeroStock_whenFindProductsWithoutZeroStock_thenCheckExistenceOfGivenProduct() {
```

Comment:
can be cut up to givenProductWithZeroStock_whenFindProducts_thenGetProducts
all other parts are obvious